pkg/querier/worker/scheduler_processor.go

```diff
@@ -222,6 +226,13 @@ func (sp *schedulerProcessor) runRequest(ctx context.Context, logger log.Logger,
 			HttpResponse: response,
 			Stats:        stats,
 		})
+		// If the used connection returned and error, remove it from the pool and retry.
+		if err != nil && retries <= maxNotifyFrontendRetries {
+			sp.frontendPool.RemoveClientFor(frontendAddress)
+			retries++
+
```

Comment:
Can we log the error (as warning) and mention retry?

Nit: As the code is right now, it will do 5 retries, ie. it will try to send notification 6 times in total.

Reply:
changed @ 12e5c31dcf5b375ff962f99249b5e4bd4333bef4